test/fgpu/test_send.py

```diff
@@ -0,0 +1,285 @@
+    sample_bits: int,
+    queues: dict[str, list[spead2.InprocQueue]],
+    send_streams: list["spead2.send.asyncio.AsyncStream"],
+    recv_streams: dict[str, list[spead2.recv.asyncio.Stream]],
+    chunks: list[Chunk],
+    sensors: aiokatcp.SensorSet,
+    time_converter: TimeConverter,
+) -> None:
+    """Test sending data via the :class:`katgpucbf.fgpu.send.Chunk` interface.
+
+    .. todo::
+
+       This is a very limited test that does not cover sensors, metrics or
+       missing data.
+    """
+    # Send descriptors to all the streams
+    descriptor_heap = make_descriptor_heap(
+        channels_per_substream=N_CHANNELS // N_SUBSTREAMS,
+        spectra_per_heap=N_SPECTRA_PER_HEAP,
+        sample_bits=sample_bits,
+    )
+    for send_stream in send_streams:
+        for substream in range(N_SUBSTREAMS):
+            await send_stream.async_send_heap(descriptor_heap, substream_index=substream)
+
+    rng = np.random.default_rng(seed=1)
+    first_timestamp = 0x123456780000
+    skip_frames = 3
+    for i, chunk in enumerate(chunks):
+        _fill_random(chunk.data, rng)
+        # Note: this doesn't correspond in any way to the values in data.
+        # That isn't necessary for this test.
+        _fill_random(chunk.saturated, rng)
+        chunk.present[:] = True
+        timestamp = first_timestamp + i * SPECTRA_SAMPLES * N_SPECTRA_PER_HEAP * N_FRAMES
+        chunk.timestamp = timestamp
+        # Check that the property works as expected
+        assert chunk.timestamp == timestamp
+    # Knock out the first few frames, to test partial transmission
+    chunks[0].present[:skip_frames] = False
+    data = np.concatenate([chunk.data for chunk in chunks])
+    saturated = np.concatenate([chunk.saturated for chunk in chunks])
+    saturated = np.sum(saturated[skip_frames:], axis=0, dtype=np.uint64)
+
+    with PromDiff(namespace=METRIC_NAMESPACE) as prom_diff:
+        # Send all the chunks, without waiting for the first one to complete
+        # transmission (to ensure that the send streams have sufficient
+        # capacity).
+        futures = [
+            asyncio.create_task(chunk.send(send_streams, N_FRAMES, time_converter, sensors, NAME)) for chunk in chunks
+        ]
+        await asyncio.gather(*futures)
+
+    for queue_list in queues.values():
+        for queue in queue_list:
+            queue.stop()
+
+    n_channels_per_substream = N_CHANNELS // N_SUBSTREAMS
+    heaps_per_iface = {iface: 0 for iface in queues.keys()}
+    for i in range(N_SUBSTREAMS):
+        seen_frames = np.zeros(N_CHUNKS * N_FRAMES, bool)
+        # Only one of the two interfaces should be receiving any data for
+        # this substream, but it's simpler to just iterate over both and
+        # find the data wherever it happened to fall.
+        for iface in queues.keys():
+            recv_stream = recv_streams[iface][i]
+            ig = spead2.ItemGroup()
+            frame = skip_frames
+            async for heap in recv_stream:
```

Comment:
We should probably move away from `heap = await stream.get()` and towards this `async for`. Certainly looks neater.

EDIT: Although, maybe it depends on the use case? 
* e.g. the `# Check that we received all the data we expected` is a good example for this usage.
* We don't necessarily want to "only receive as much data as we know we sent".